main.py

```diff
@@ -798,6 +798,29 @@ def delete_link(self, request: Request) -> JSONResponse:
         self.controller.buffers.app.put(event)
         return JSONResponse("Operation successful")
 
+    @rest('v3/interfaces/{intf_id}', methods=['DELETE'])
+    def delete_interface(self, request: Request) -> JSONResponse:
+        """Delete an interface only if it is not used."""
+        intf_id = request.path_params.get("intf_id")
+        intf_split = intf_id.split(":")
+        switch_id = ":".join(intf_split[:-1])
+        intf_port = int(intf_split[-1])
```

Comment:
This will crash if an invalid wrong is passed such as `00:00:00:00:00:00:00:01:x`. 

The potential exc needs to be handled.

Reply:
Added `try except` to catch  no numeric ports